[Feature] Add highlighted messages sound notification

whitecrownclown opened the pull request:
- Added sound notification for chat mention
- Added setting for this (default false)

Fixes #3338

whitecrownclown commented:
@night Hello, would you kindly have a look? I've also added the mp3 file (audio notification, can be changed of course) and i assume it needs to be manually uploaded to the cdn (not sure about this). Thanks!

whitecrownclown commented:
@night Yes, it works, i've tested it on both firefox and chrome. I've also moved this into the highlight module.

night commented on src/modules/chat_highlight_blacklist_keywords/index.js:
```diff
@@ -184,6 +186,55 @@ class ChatHighlightBlacklistKeywordsModule {
+        });
+
+        this.username = twitch.getChatServiceClient().configuration.username.toLowerCase();
+        this.soundHighlight = false;
```
why are we storing this as a property of the class? why not just check the settings?

whitecrownclown replied:
Updated.

night commented on src/modules/chat_highlight_blacklist_keywords/index.js:
```diff
@@ -184,6 +186,55 @@ class ChatHighlightBlacklistKeywordsModule {
+    }
+
+    handleSoundHighlight(messageParts = []) {
+        const isMentioned = messageParts.some(({content}) => {
```
why don't we just play the sound from at line 271 and skip all this "mentioned" logic?

whitecrownclown replied:
So scrap the mentioned logic and add this for highlights instead?

night commented on src/modules/chat_highlight_blacklist_keywords/index.js:
```diff
@@ -184,6 +186,55 @@ class ChatHighlightBlacklistKeywordsModule {
+        if (node.length) {
+            node[0].play();
+        } else {
+            node = document.createElement('audio');
```
why do we need to make an audio element? can't we just load it in js and play it in js?

night commented on src/modules/chat_highlight_blacklist_keywords/index.js:
```diff
@@ -184,6 +186,55 @@ class ChatHighlightBlacklistKeywordsModule {
+            node.hidden = 'hidden';
+            node.id = CHAT_MENTION_NOTIFICATION_ID;
+            node.autoplay = true;
+            node.src = cdn.url('assets/notification.mp3');
```
we have a `tink` sound someone in git history that we should use instead

whitecrownclown commented:
@night 	A summary of all the changes:

- Changed notification sound to `tink` (got it by searching through git history);
- Removed `mentioned` logic in favor of playing the notification for `highlighted` messages.
